### ISSUE TITLE
Upgrader causes database errors

### ISSUE THREAD

**Issue by qstokkink** [Tribler/tribler#8162]
Our upgrade script uses `iterdump` to get the scheme and content of an existing database.

https://github.com/Tribler/tribler/blob/b8b2563898194b601ddc9d05f92da41a009e8168/src/tribler/upgrade_script.py#L104

Then, if an existing database is found, the lines of the dump are sequentially executed to inject the old data into the new database:

https://github.com/Tribler/tribler/blob/b8b2563898194b601ddc9d05f92da41a009e8168/src/tribler/upgrade_script.py#L110-L114

This does exactly that. However, the content that is included in this dump also **includes ids**, for example:

```
INSERT INTO "Peer" VALUES(1,X'...','2024-01-05 13:09:23.894782');
INSERT INTO "Peer" VALUES(2,X'...','2024-01-05 13:11:05.888214');
INSERT INTO "Peer" VALUES(3,X'...','2024-01-05 13:11:05.901082');
INSERT INTO "Peer" VALUES(4,X'...','2024-01-05 13:11:10.887803');
```

This is where the approach becomes less fine: if an id already exists, this imported data gets dropped. However, it gets worse: if another entry that points to this id _does get imported_ in another table, it may point to a completely different object with the same id. I believe this causes errors like #8111.

---

In short, I think we need to change our approach and reconstruct objects from the existing data. Instead of copying the data, we would create actual new objects (in the orm) from the old data. The downside is that _this is much slower_ than our blind `iterdump` approach.

**Comment by qstokkink** [comment]
[WIP] Reconstruction SQL:

## StatementOP

```sql
SELECT SubjectResource.name, SubjectResource.type, ObjectResource.name, ObjectResource.type, Statement.added_count, Statement.removed_count, Statement.local_operation, Peer.public_key, Peer.added_at, StatementOp.operation, StatementOp.clock, StatementOp.signature, StatementOp.updated_at, StatementOp.auto_generated
FROM StatementOp
INNER JOIN Peer ON StatementOp.peer=Peer.id
INNER JOIN Statement ON StatementOp.statement=Statement.id
INNER JOIN Resource AS SubjectResource ON Statement.subject=SubjectResource.id
INNER JOIN Resource AS ObjectResource ON Statement.object=ObjectResource.id
;
```
